### ISSUE TITLE
Make reasonable error message when user reuses a process name

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#14402]
If a user uses a process name which is already in the edm ROOT file then we fail with a poor error message when trying to resolve consumes for a module which used an empty string for the process name in the consumes specification:

```
----- Begin Fatal Exception 06-May-2016 19:04:57 CEST-----------------------
An exception of category 'EDConsumerBase' occurred while
  [0] Calling beginJob
Exception Message:
insertFoundModuleLabel Couldn't find ModuleDescription for the consumed module label: hbherecoPedestal
----- End Fatal Exception -------------------------------------------------
```

We should detect the problem sooner and issue a helpful error message. We also need a unit test to make sure we do not regress.

**Comment by Dr15Jones** [comment]
assign core

**Comment by wddgit** [comment]
This is fixed by PR #15309

**Comment by Dr15Jones** [comment]
+1
